mikeio1d/res1d.py

```diff
@@ -107,19 +112,37 @@ class Res1D:
 
     def __init__(
         self,
-        file_path=None,
-        lazy_load=False,
-        header_load=False,
-        reaches=None,
-        nodes=None,
-        catchments=None,
-        time=None,
-        col_name_delimiter=NAME_DELIMITER,
-        put_chainage_in_col_name=True,
-        clear_queue_after_reading=True,
-        result_reader_type=ResultReaderType.COPIER,
-        derived_quantities=None,
+        file_path: Union[str, Path] = None,
+        reaches: Optional[list[str]] = None,
+        nodes: Optional[list[str]] = None,
+        catchments: Optional[list[str]] = None,
+        time: Union[tuple[str], list[str], slice, None] = None,
+        derived_quantities: Optional[list[str]] = None,
+        **kwargs,
     ):
+        # region deprecation
+
+        def warn_deprecation(name: str, hint: str = ""):
+            if name in kwargs:
+                warnings.warn(f"The '{name}' parameter will be deprecated in 1.0. {hint}", FutureWarning)
+
+        warn_deprecation("lazy_load")
+        warn_deprecation("col_name_delimiter")
+        warn_deprecation("put_chainage_in_col_name")
+        warn_deprecation("clear_queue_after_reading")
+        warn_deprecation("header_load", "Dynamic data is read lazily, so header_load is not needed.")
+        warn_deprecation("result_reader_type", "Use res1d.result_reader_type instead.")
+
+        lazy_load = kwargs.get("lazy_load", False)
+        col_name_delimiter = kwargs.get("col_name_delimiter", NAME_DELIMITER)
+        put_chainage_in_col_name = kwargs.get("put_chainage_in_col_name", True)
+        clear_queue_after_reading = kwargs.get("clear_queue_after_reading", True)
+        header_load = kwargs.get("header_load", False)
+        result_reader_type = kwargs.get("result_reader_type", ResultReaderType.COPIER)
```

Comment:
I thought we agreed that this one we do not deprecate :-).

Reply:
I can't recall 🙈 It's still possible to change the reader via Res1D.reader = ResultReaderCreator.create(...). I'm in favor of as clean as possible of a parameter set for the public scripting API. Unless you're using it very often in the constructor, then I'm inclined to remove it. What do you think? @gedaskir